app/components/post_body/post_body.js

```diff
@@ -348,6 +349,13 @@ export default class PostBody extends PureComponent {
         const messageStyle = isSystemMessage ? [style.message, style.systemMessage] : style.message;
         const isPendingOrFailedPost = isPending || isFailed;
 
+        const messageStyles = {messageStyle, textStyles};
+        const intl = this.context.intl;
+        const systemMessage = renderSystemMessage(this.props, messageStyles, intl);
```

Comment:
this is being run for every post. Can you just render the system message if the post is actually a system message `isSystemMessage` prop and make sure is not a combined system message `postType !== Posts.POST_TYPES.COMBINED_USER_ACTIVITY`

Reply:
I don't think `isSystemMessage` can be used here because we won't have a renderer for every system message type like ephemeral messages or anything new that gets added later on.

I could go either way on whether `PostBody` or `renderSystemMessage` has to check what message types `renderSystemMessage` can handle, but this method does match how the web app handles it

Reply:
You're right. This interface took me some digging too because of the variety of 'system' message flavours I unearthed in the Post body API. Ultimately, I decided to mimic web app behaviour to keep implementation consistent for now. Then a future redesign/refactor would apply the same way to mobile and web.